src/App.tsx

```diff
@@ -15,36 +15,81 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
 export const App: React.FC = () => {
+  const [isReversed, setIsReversed] = useState(false);
+  const [sortType, setSortType] = useState(SortType.NONE);
+
+  function getReorderedGoods(goods: string[]) {
+    const visibleGoods = [...goods];
+
+    if (sortType === SortType.ALPHABET) {
+      visibleGoods.sort((a: string, b: string) => a.localeCompare(b));
+    } else if (sortType === SortType.LENGTH) {
+      visibleGoods.sort((a: string, b: string) => a.length - b.length);
+    }
+
+    if (isReversed) {
+      visibleGoods.reverse();
+    }
+
+    return visibleGoods;
+  }
+
+  const reorderedGoods = getReorderedGoods(goodsFromServer);
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={`button is-info${sortType !== SortType.ALPHABET ? ' is-light' : ''}`}
+          onClick={() => setSortType(SortType.ALPHABET)}
+        >
           Sort alphabetically
         </button>
 
-        <button type="button" className="button is-success is-light">
+        <button
+          type="button"
+          className={`button is-success${sortType !== SortType.LENGTH ? ' is-light' : ''}`}
+          onClick={() => setSortType(SortType.LENGTH)}
+        >
           Sort by length
         </button>
 
-        <button type="button" className="button is-warning is-light">
+        <button
+          type="button"
+          className={`button is-warning${!isReversed ? ' is-light' : ''}`}
+          onClick={() => setIsReversed(value => !value)}
+        >
           Reverse
         </button>
 
-        <button type="button" className="button is-danger is-light">
-          Reset
-        </button>
+        {(sortType !== SortType.NONE || isReversed) && (
```

Comment:
Here you have just used a check, and at your previous PR you have created a function for it. What has changed? :) 

In my opinion, the best approach is in the middle: you should just declare a variable for it. 
`const resetButtonDisplayed(or something) = sortType !== SortType.NONE || isReversed`